internal/storage/cp/cp.go

```diff
@@ -0,0 +1,138 @@
+		}
+		// TODO: Check if destination is a directory
+		return client.UploadStorageObject(ctx, projectRef, dstParsed.Path, src, fsys)
+	} else if strings.ToLower(srcParsed.Scheme) == ls.STORAGE_SCHEME && strings.ToLower(dstParsed.Scheme) == ls.STORAGE_SCHEME {
+		return errors.New("Copying between buckets is not supported")
+	}
+	utils.CmdSuggestion = fmt.Sprintf("Run %s to copy between local directories.", utils.Aqua("cp -r <src> <dst>"))
+	return errUnsupportedOperation
+}
+
+func DownloadStorageObjectAll(ctx context.Context, projectRef, remotePath, localPath string, fsys afero.Fs) error {
+	// Prepare local directory for download
+	if fi, err := fsys.Stat(localPath); err == nil && fi.IsDir() {
+		localPath = filepath.Join(localPath, path.Base(remotePath))
+	}
+	count := 0
+	if err := ls.IterateStoragePathsAll(ctx, projectRef, remotePath, func(objectPath string) error {
+		relPath := strings.TrimPrefix(objectPath, remotePath)
+		dstPath := filepath.Join(localPath, filepath.FromSlash(relPath))
+		fmt.Fprintln(os.Stderr, "Downloading:", objectPath, "=>", dstPath)
+		count++
+		if strings.HasSuffix(objectPath, "/") {
+			return utils.MkdirIfNotExistFS(fsys, dstPath)
+		}
+		if err := utils.MkdirIfNotExistFS(fsys, filepath.Dir(dstPath)); err != nil {
+			return err
+		}
+		return client.DownloadStorageObject(ctx, projectRef, objectPath, dstPath, fsys)
+	}); err != nil {
+		return err
+	}
+	if count == 0 {
+		return errors.New("Object not found: " + remotePath)
+	}
+	return nil
+}
+
+func UploadStorageObjectAll(ctx context.Context, projectRef, remotePath, localPath string, fsys afero.Fs) error {
+	noSlash := strings.TrimSuffix(remotePath, "/")
+	// Check if directory exists on remote
+	dirExists := false
+	fileExists := false
+	if err := ls.IterateStoragePaths(ctx, projectRef, noSlash, func(objectName string) error {
+		if objectName == path.Base(noSlash) {
+			fileExists = true
+		}
+		if objectName == path.Base(noSlash)+"/" {
+			dirExists = true
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	baseName := filepath.Base(localPath)
+	return afero.Walk(fsys, localPath, func(filePath string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		relPath, err := filepath.Rel(localPath, filePath)
+		if err != nil {
+			return err
+		}
+		dstPath := remotePath
+		// Copying single file
+		if relPath == "." {
+			if IsDir(dstPath) || (dirExists && !fileExists) {
+				dstPath = path.Join(dstPath, info.Name())
+			}
+		} else {
+			if baseName != "." && (dirExists || len(noSlash) == 0) {
+				dstPath = path.Join(dstPath, baseName)
+			}
+			dstPath = path.Join(dstPath, relPath)
+		}
+		fmt.Fprintln(os.Stderr, "Uploading:", filePath, "=>", dstPath)
+		err = client.UploadStorageObject(ctx, projectRef, dstPath, filePath, fsys)
+		if err != nil && strings.Contains(err.Error(), `"error":"Bucket not found"`) {
```

Comment:
I'm not sure if we should always create a bucket without any additional flag, but from UX perspective it sounds ok.